codegen/src/codegen.ts

```diff
@@ -20,10 +20,16 @@ function getTypeName(tableName: string) {
 
 function generateTableType(table: Table) {
   const { columns } = table;
-  const revLinks: { table: string }[] = []; // table.rev_links || [];
+  const revLinks: { table: string }[] = [];
+
+  const properties = [
+    ...columns.map((column) => generateColumnType(column)),
+    ...revLinks.map((link) => `${link.table}: Query<${getTypeName(link.table)}>`)
+  ];
+
   return `export interface ${getTypeName(table.name)} {
-    ${columns.map((column) => generateColumnType(column)).join('\n')}
-    ${revLinks.map((link) => `${link.table}: Query<${getTypeName(link.table)}>`).join('\n')}
+    ${properties.join('\n')}
+    [key: string]: unknown;
```

Comment:
Does this solve the circular references?

The problem is that now users won't have compilation errors if they do `obj.columnThatDoesNotExist`, right?

Reply:
> Does this solve the circular references?

Yes, it's a TS long living error (they attempted to fix it in 3.7 but you still need a named index access to make them work)

> The problem is that now users won't have compilation errors if they do obj.columnThatDoesNotExist, right?

That's right, it will return "unknown" type but no compile error.

That's why utils like zod have ``lazy`` initializers and sometimes they're unable to type themselves with circular references. It can be solved but with a tradeoff.